### PR TITLE
#2845 add mobile version header

### LINKED ISSUE #2845: Add mobile version header
## Is your feature request related to a problem? Please describe.

Hotfix for `v5.0.1` to fix date parsing bug.

## Describe the solution you'd like

Add mobile version header to requests.

## Implementation

Just piggyback onto mobile hardware UUID logic. 

## Describe alternatives you've considered

N/A.

## Additional context

See https://github.com/sussol/msupply/pull/6230.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #2845.

## Change summary

Hotfix for `v5.0.2`.

### Setup

- Get version header desktop structure [here](https://github.com/sussol/msupply/pull/6230).

### Test cases 

- [ ] Initial sync with `v4.10` occurs without error.
- [ ] Initial sync with `v4.11` occurs without error.
- [ ] Initial sync with `#6230` occurs without error.
- [ ] On sync with `v4.10`, incoming dates are formatted as ISO **GMT** (see note below).
- [ ] On sync with `v4.11`, incoming dates are formatted as ISO **GMT** (see note below).
- [ ] On sync with `#6230`, incoming dates are formatted as ISO **GMT** (see note below).

Note: log `parseDate` inputs and check for the timezone offset, e.g. `"Z"` (not sure how this can be tested in production!)

### Related areas to think about

N/A.

### CODE REVIEW

File: src/sync/Synchroniser.js [hunk 1]
```diff
@@ -57,7 +57,10 @@ export class Synchroniser {
     this.settings = settings;
     this.syncQueue = new SyncQueue(this.database);
     this.dispatch = dispatch;
-    this.extraHeaders = { 'msupply-site-uuid': DeviceInfo.getUniqueId() };
+    this.extraHeaders = {
+      'msupply-site-uuid': DeviceInfo.getUniqueId(),
+      'mobile-version': '5.0.0',
```
Comment (andreievg) [inline src/sync/Synchroniser.js:62]: should come from package.json, i.e. :https://github.com/openmsupply/mobile/blob/dfa74328694a98858568f0eb6e5e775eb9f81561/src/dataMigration.js#L33

Comment (wlthomson) [reply src/sync/Synchroniser.js]: Yup, agreed.

864871feadfe5ee636f3a394ef52643853d26525.

Comment (andreievg) [reply src/sync/Synchroniser.js]: @wlthomson, shouldn't be merging an approved PRs
You also missed a console.log

Comment (wlthomson) [reply src/sync/Synchroniser.js]: @andreievg Merged into release candidate due to time pressures. Main PR is still open: #2854.

**Pull request merged**